### PR TITLE
feat: Implement named custom filters 

### LINKED ISSUE #18823: Persist custom filters via `localStorage`
### Description

Instead of https://github.com/camunda/camunda/issues/17716 we'll persist the custom filters via `localStorage`

### TODO

- [ ] Add option of setting a name to a filter
- [ ] Persist it via `localStorage`

### Additional Information

---------------------------------------------------------------------------------------------

### Acceptance Criteria

- [ ] Checked if the specific data needs to be provided for the QA process

### Definition of Ready - Checklist

- [ ] The issue has a meaningful title, description and testable acceptance criteria

### PR DESCRIPTION
**Pull request by vsgoulart**
## Description

Please review https://github.com/camunda/camunda/pull/18848 first

This PR creates the option of creating persistable named filters

## Related issues

closes #18823

### CODE REVIEW

**Comment by vsgoulart** [comment]
Tests for this one will come later since this PR is already too big

**Review by marcosgvieira — APPROVED** [review]
LGTM!

File: tasklist/client/src/Tasks/CollapsiblePanel/CustomFiltersModal/index.tsx [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+  filterId,
+  isOpen,
+  onSuccess,
+  onDelete,
+  ...props
+}) => {
+  const [customFilters, setCustomFilters] = useState<CustomFilters>(
+    DEFAULT_FILTER_VALUES,
+  );
+  const [currentStep, setCurrentStop] = useState<'fields' | 'name' | 'delete'>(
```
Comment (Skaiir) [inline tasklist/client/src/Tasks/CollapsiblePanel/CustomFiltersModal/index.tsx:67]: Lil typo but it did confuse me in reading.

**Review by Skaiir — COMMENTED** [review]
Honestly code looks good, found a couple things in the early commits that were caught later. Aside from that just some minor stuff. Nice one :)

**Pull request merged**